### PR TITLE
[stable/magic-namespace] initial commit

### PR DESCRIPTION
**Pull request by krancour**
See `README.md` for explanation of what this chart does.

Further reading: https://engineering.bitnami.com/articles/helm-security.html

cc @technosophos @michelleN, this is what I was planning to show you both on Monday.

### CODE REVIEW

**Comment by unguiculus** [comment]
This is cool stuff. Thanks for this.

cc @kubernetes/charts-maintainers

File: incubator/magic-namespace/README.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+- A Kubernetes cluster with RBAC enabled
+
+## Installing the Chart
+
+To install the chart to create the `foo` namespace (if it doesn't already exist)
+and useful resources (Tiller, service accounts, etc.) within that namespace:
+
+```bash
+$ helm install incubator/magic-namespaxce --name foo --namespace foo
```
Comment (technosophos) [inline incubator/magic-namespace/README.md]: I think namespaxce is a typo.

Comment (krancour) [reply incubator/magic-namespace/README.md]: Thanks @technosophos. Fixed!

**Comment by technosophos** [comment]
Just saw an interactive demo of this. It is _very_ cool.

**Comment by prydonius** [comment]
This is pretty sweet, thanks @krancour. Hate to be a killjoy, but I wonder if this would benefit from a more descriptive name (e.g. `tiller-multitenant` or `tiller-namespaced`)?

**Comment by krancour** [comment]
> I wonder if this would benefit from a more descriptive name (e.g. tiller-multitenant or tiller-namespaced

@prydonius good question. My real goal here is for this to be a comprehensive option for cluster operators to manage _namespaces_ in a sane fashion using a familiar Helm-based workflow (instead of scripts). The fact that it (by default) installs a Tiller-per-namespace is but one facet of that. In light of this, I think it's better if the name doesn't over-emphasize Tiller.

Having said all that, I have no special attachment to the current name and I remain quite open to alternatives.

**Comment by krancour** [comment]
Any suggestion on who to assign this to?

File: incubator/magic-namespace/templates/rolebindings.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+
+{{- $ctx := . }}
```
Comment (prydonius) [inline incubator/magic-namespace/templates/rolebindings.yaml]: IIRC, you can use `$` within a `range` to point to the outer context, no need to create a variable.

File: incubator/magic-namespace/templates/serviceaccounts.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{{- $ctx := . }}
```
Comment (prydonius) [inline incubator/magic-namespace/templates/serviceaccounts.yaml]: see above

**Review by prydonius — APPROVED** [review]
minor comments, mostly lgtm

/ok-to-test

**Comment by prydonius** [comment]
This could probably go straight to stable, it's in a good state IMO. WDYT @unguiculus @technosophos @krancour?

**Comment by unguiculus** [comment]
Yes, definitely. 👍

File: incubator/magic-namespace/templates/serviceaccounts.yaml (hunk 3 above)
Comment (krancour) [reply incubator/magic-namespace/templates/serviceaccounts.yaml]: I'll try that.

**Comment by krancour** [comment]
@prydonius, I'm totally fine with that. I'll amend the PR accordingly. I just hand't wanted to jump the gun. Thanks for thinking this is stable-worthy.

**Comment by krancour** [comment]
@prydonius feedback has been addressed-- moved chart from `incubator/` to `stable/` and also you were right that `$` works as a fine replacement for `.` when in a loop and there is no need to declare a separate variable for that purpose.

**Review by prydonius — APPROVED** [review]
/lgtm

**Pull request merged**